src/main/java/com/kodilla/ecommercee/dto/OrderDto.java

```diff
@@ -0,0 +1,20 @@
+package com.kodilla.ecommercee.dto;
+
+
+import com.kodilla.ecommercee.domain.Product;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import java.time.LocalDate;
+import java.util.List;
+
+@AllArgsConstructor
+@Getter
+public class OrderDto {
+
+    private Long id;
+    private LocalDate date;
+    private List<Product> productList;
+
```

Comment:
Uboga ta klasa, brakuje mi  przynajmniej pól userId, cartId.

Reply:
racja dodalem usera, i zmienilem liste na hashmape, ale nie widze sensu carta, poniewaz wg mnie cart po złożeniu zamówienia się czyści i wszystko jest przechowywane w zamówieniu, tzn. w danym momencie istnieje tylko jeden cart, a nie wiem czy grupy produktów mają jakieś znaczenie tutaj

Reply:
Dodałeś całą klasę user gdzie mogą być przechowywane dane wrażliwe, zresztą nie potrzebujemy dostępu do wszystkich informacji usera. Myślę że śmiało możemy użyć tylko Id  Proponuje wyciągnąć tylko Long  userId.

Zamiast HashMapy użyłbym LinkedHashMap  pozwoli to wyświetlać koszyk w kolejności takiej jak zrobił to user.

Co do cartId to chyba faktycznie będzie zbędne.

Reply:
Dobry pomysł z linked, ale z user bede potrzebowal takich rzeczy jak adres i dane kontaktowe. Zamówienie nie jest publicznie dostępne, tylko między klientem a sklep więc chyba moge po prostu trzymac tam usera i pobierac dane prosto z niego, niż trzymac id, uszkac usera z danym id i wtedy pobierać.

Reply:
Dobry pomysł z linked, ale z user bede potrzebowal takich rzeczy jak adres i dane kontaktowe. Zamówienie nie jest publicznie dostępne, tylko między klientem a sklep więc chyba moge po prostu trzymac tam usera i pobierac dane prosto z niego, niż trzymac id, uszkac usera z danym id i wtedy pobierać.

Reply:
Nie wiemy jakie dane będą przechowywane w User. 
Załóżmy że będzie tam hasło, jakiś token.  
Nie wiedząc co kryje się się w tej klasie chcesz ją cała wyrzucić do jsona?

niech ktoś się jeszcze wypowie w tym temacie,  nie jestem do końca przekonany że wiem co mówię  😆

Reply:
Wydaje sie sensowne. Wieczorem zmienie.

Reply:
Alternatywą może byc stworzenia klasy UserOrderData oznaczoną adnotacja @JsonIgnoreProperty(ignoreUnknown = true) posiadajacą pola adresu zamieszkania , kontaktu, generalnie dane które nie są wrażliwe i potrzebne do realizacji zamówienia. Dzięki tej adnotacji Spring przypisze tworzonemu obiektowi tylko te pola które będzie zawierać ta klasa (UserOrderData), co oznacza ,że nie znajdą się w tym obiekcie dane wrażliwe , a jedynie dane  potrzebne do zrealizowania zamówienia. I taką klase możesz bez problemu przekazać jako pole do klasy OrderDto.

Reply:
Czyli , jeżeli ktoś kliknie "Create order" to wtedy na podstawie klasu User będzie sie tworzyć obiekt klasy UserOrderData , której pola wykorzystamy do realizacji zamówienia  :D

Reply:
Mam pewną uwagę:
- pole productList reprezentuje  zbiór produktów danego zamówienia ? Jeżeli tak to co w przypadku gdy osoba będzie chiała kupić dwa takie same produkty ? HashMapa nie pozwala na posiadanie dwóch takich samych kluczy. Co za tym idzie , w momencie gdy ktos będzie chciał dodać taki sam produkt to hashMapa to zignoruje i go nie doda.
Czyli klucz tej hash mapy to produkt a jej wartosć to ? cena ? jeśli sie myle to daj znać :)

Reply:
klucz to produkt, wartość to ilość, cenna powinna być polem produktu

Reply:
aha okej w takim razie HashMapa będzie sie sprawować :)

Reply:
.put(key, value) aktualizuje value danego klucza a jeśli takiego nie ma to go dodaje, więc nie będzie problemu jeśli ktoś będzie chciał coś dokupić

Reply:
dodaje?  a nie aktualizuję?    
 Map<String,Integer> m = new HashMap<>();
m.put("Test",1);
m.put("Test",1);
 System.out.print(m.get("Test")); 
wynik:
[1]

Jeżeli użyję addItemToCard  raz jeszcze to mi nie doda tego tylko znowu nadpiszę 1,  gdybyśmy przesyłali parametrem ilość to by nadpisał poprawną. Lecz wychodzi na to że według specyfikacji nie ma miejsca na pole ilość sztuk  produkt powinien wyświetlić się ewentualnie dwa razy.

Myślę że lista byłaby tu odpowiednia.